VideoWeb/VideoWeb/ClientApp/src/app/services/conference/conference.service.ts

```diff
@@ -0,0 +1,164 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, NavigationEnd, ParamMap, Router } from '@angular/router';
+import { Guid } from 'guid-typescript';
+import { BehaviorSubject, Observable, ReplaySubject, Subscription } from 'rxjs';
+import { filter, map, mergeMap, take } from 'rxjs/operators';
+import { ParticipantModel } from 'src/app/shared/models/participant';
+import { ApiClient, ConferenceResponse, ConferenceStatus } from '../clients/api-client';
+import { EventsService } from '../events.service';
+import { LoggerService } from '../logging/logger.service';
+import { ConferenceStatusMessage } from '../models/conference-status-message';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ConferenceService {
+    private loggerPrefix = '[ConferenceService] -';
+
+    private subscriptions: Subscription[] = [];
+    constructor(
+        router: Router,
+        private activatedRoute: ActivatedRoute,
+        private eventService: EventsService,
+        private apiClient: ApiClient,
+        private logger: LoggerService
+    ) {
+        logger.conferenceService = this;
+        router.events
+            .pipe(
+                filter(x => x instanceof NavigationEnd),
+                map(() => activatedRoute.snapshot),
+                map(route => {
+                    while (route && !route.paramMap?.has('conferenceId')) {
+                        route = route?.firstChild;
+                    }
+
+                    return route?.paramMap;
+                })
+            )
+            .subscribe(paramMap => {
+                this.onRouteParamsChanged(paramMap);
+            });
+    }
+
+    private _currentConference: ConferenceResponse;
+    get currentConference(): ConferenceResponse {
+        return this._currentConference;
+    }
+
+    private currentConferenceSubject: ReplaySubject<ConferenceResponse> = new ReplaySubject<ConferenceResponse>();
+    get currentConference$(): Observable<ConferenceResponse> {
+        return this.currentConferenceSubject.asObservable();
+    }
+
+    private onCurrentConferenceStatusChangedSubject: BehaviorSubject<{
+        oldStatus: ConferenceStatus;
+        newStatus: ConferenceStatus;
+    }> = new BehaviorSubject<{ oldStatus: ConferenceStatus; newStatus: ConferenceStatus }>({
+        oldStatus: this.currentConference?.status,
+        newStatus: null
+    });
```

Comment:
Could we use a ReplaySubject<ConferenceStatus>(2) here?

Reply:
Actually I don't know how returning a ReplaySubject as an Observable works.

Reply:
I would prefer to do it this way but it would stop the initial subscribe as it would only emit after the second value? I will have a look and see if there is a way to combine them.